### ISSUE TITLE
Don't require Clojure dependency

### ISSUE THREAD

**Issue by efge** [michaelklishin/quartz-mongodb#129]
Would you consider a patch to make Clojure an optional dependency? It's probably required in the original authors' use cases, but for most of us it's an additional unneeded 3.7 MB JAR.

I'm thinking of making `DynamicClassLoadHelper` test the presence of `clojure.lang.DynamicClassLoader` by reflection, and if not present just delegate to the standard `CascadingClassLoadHelper`.

**Comment by michaelklishin** [comment]
DynamicMongoDBJobStore can be moved to Quartzite.

**Comment by michaelklishin** [comment]
I evaluated moving `DynamicMongoDBJobStore` to Quartzite. Arguably it fits here and for reasons that are not very interesting, it is easier to make Clojure an optional dependency here than in Leiningen (but retain the ability for it to compile Java sources).

Clojure already was a `provided` dependency, I switched to a [compile only](https://blog.gradle.org/introducing-compile-only-dependencies) one.
